staking/src/lib.rs

```diff
@@ -0,0 +1,249 @@
+#![no_std]
+
+use codec::{Decode, Encode};
+use ft_io::*;
+use gstd::{exec, msg, prelude::*, ActorId};
+use scale_info::TypeInfo;
+use staking_io::*;
+
+#[derive(Debug, Default, Encode, Decode, TypeInfo)]
+struct Staking {
+    staking_token_address: ActorId,
+    reward_token_address: Option<ActorId>,
+    tokens_per_stake: u128,
+    total_staked: u128,
+    distribution_time: u64,
+    produced_time: u64,
```

Comment:
Because there no reward update function

Reply:
> Because there no reward update function

`update_reward(&mut self)` here it is, and is called very often

Reply:
> > Because there no reward update function
> 
> `update_reward(&mut self)` here it is, and is called very often

Louise was referring to the setRewardTotal() function